### ISSUE TITLE
svg-sprite: more flexibility for generated icon components

### ISSUE THREAD

**Issue by fweinaug** [kiliman/rmx-cli#21]
i noticed that when using the `--components` flag the generated icon components assume that there is a Icon component and that it has a icon prop. also they only provide the SVG element props. this does not work well when using a template.

1. a quick but incomplete solution (that would solve my use case) could be this, but it still assumes the component name and icon prop name:
```
      output += `
export const ${componentName}Icon = (props: Omit<React.ComponentProps<typeof Icon>, 'icon'>) => <Icon icon="${icon}" {...props} />;`
```

2. a better solution, but with more effort, could be to allow a template also for the generated components.

what do you think @kiliman ?

**Comment by fweinaug** [comment]
currently the generated icon components look like this:
`export const AddIcon = (props: SVGProps<SVGSVGElement>) => <Icon icon="add" {...props} />;`

i would like to use `name` instead of `icon` and the props type should be different (it has `size` for example). i could create a PR that adds a new `component-template` option. in the template you would use `%ICON_NAME%` to inject the name.

what do you think @kiliman ?

**Comment by kiliman** [comment]
Fixed in v0.4.15
